### PR TITLE
Add websocket server

### PR DESCRIPTION
**Pull request by aninternetof**
This adds websocket endpoints for use by Mission Control and the rover.

To try it out, visit `http://localhost:8000/realtime/foobar` in two different browser tabs. Enter a message in one, and you should see it in the other.
I left these views in because I figure they will be helpful for debugging.

Note that the `foobar` above is a url parameter -- it could be anything. In actual use, it will probably be the OAuth client-id of the rover.

Also, here is a samply python websocket client to test with it: https://gist.github.com/aninternetof/de48feef6baef7d7d2f0667334297724

This knocks our testing coverage down to 99%, but per [this conversation](https://github.com/django/channels/issues/1045), I couldn't find a meaningful way to test realtime/routing.py.

Still to-do:

- Check that the user owns the rover when he tries to connect to the websocket endpoint of that rover's clientId. I've already included the AuthMiddleware around the websocket router, so the user information should be ready-to-go in the `context` variable of the consumer.
- Decide what the messages should look like! Maybe something like `set left_motor 100` or `sensor left_ir 1`?

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [pytest]
-DJANGO_SETTINGS_MODULE=config.settings.local
+DJANGO_SETTINGS_MODULE=config.settings.test
```
Comment (aninternetof) [inline pytest.ini]: Not sure why this wasn't `test` already.

File: compose/django/daphne.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+python /app/manage.py migrate
+python /app/manage.py collectstatic --noinput
+/usr/local/bin/daphne -b 0.0.0.0 -p 5000 config.asgi:application
```
Comment (aninternetof) [inline compose/django/daphne.sh]: `daphne` is Django's dual HTTPS and WebSocket server. It is production speed.

File: compose/django/Dockerfile-dev [hunk 3]
```diff
@@ -10,7 +10,8 @@ RUN apk add --no-cache bash ca-certificates wget build-base python3-dev postgres
     && update-ca-certificates \
     && pip install pipenv \
     && pipenv install --dev --system \
-    && apk del --purge python3-dev
+    && apk del --purge python3-dev \
+    && pip freeze
```
Comment (cabarnes) [inline compose/django/Dockerfile-dev]: Why is `pip freeze` needed? `pipenv` should be handling the dependencies

File: compose/django/gunicorn.sh [hunk 4]
```diff
@@ -1,4 +1,4 @@
 #!/bin/sh
 python /app/manage.py migrate
 python /app/manage.py collectstatic --noinput
-/usr/local/bin/gunicorn config.wsgi -w 4 -b 0.0.0.0:5000 --chdir=/app
\ No newline at end of file
+/usr/local/bin/gunicorn config.wsgi -w 4 -b 0.0.0.0:5000 --chdir=/app
```
Comment (cabarnes) [inline compose/django/gunicorn.sh]: Is this file needed anymore?

File: Pipfile [hunk 5]
```diff
@@ -51,3 +52,5 @@ django-rest-auth = "==0.9.3"
 djangorestframework-jwt = "==1.11.0"
 coreapi = "==2.3.3"
 markdown = "==2.6.11"
+channels = "==2.1.3"
+channels_redis = "==2.3.0"
```
Comment (cabarnes) [inline Pipfile]: Shouldn't `daphne` be in this list? Also, should `gunicorn` be removed?

File: compose/django/start-dev.sh [hunk 6]
```diff
@@ -1,3 +1,3 @@
 #!/bin/sh
 python manage.py migrate
-python manage.py runserver_plus 0.0.0.0:8000
+python manage.py runserver 0.0.0.0:8000
```
Comment (cabarnes) [inline compose/django/start-dev.sh]: `runserver_plus` came from `django-extensions`. Was there an issue using it?

File: config/asgi.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+application = ProtocolTypeRouter({
+    # (http->django views is added by default)
+    'websocket': AuthMiddlewareStack(
+        URLRouter(
+            realtime.routing.websocket_urlpatterns
+        )
+    ),
+})
+
+if os.environ.get('DJANGO_SETTINGS_MODULE') == 'config.settings.production':
```
Comment (cabarnes) [inline config/asgi.py]: It looks like `DJANGO_SETTINGS_MODULE` was hardcoded to `config.settings.local` above so I don't think this can ever happen

File: compose/django/Dockerfile-dev (hunk 3 above)
Comment (aninternetof) [reply compose/django/Dockerfile-dev]: Sorry, that was a debugging addition. Removing.

File: compose/django/gunicorn.sh (hunk 4 above)
Comment (aninternetof) [reply compose/django/gunicorn.sh]: Nope! Removing.

File: Pipfile (hunk 5 above)
Comment (aninternetof) [reply Pipfile]: Yes and yes.

File: config/asgi.py (hunk 7 above)
Comment (aninternetof) [reply config/asgi.py]: Whoops. Removed hardcoded line above.

File: compose/django/start-dev.sh (hunk 6 above)
Comment (aninternetof) [reply compose/django/start-dev.sh]: `runserver` has an asgi (websockets) server built in that hasn't made it into `runserver_plus` yet.

File: compose/django/daphne.sh [hunk 8]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+export DJANGO_SETTINGS_MODULE=config.settings.production
```
Comment (cabarnes) [inline compose/django/daphne.sh]: This should already be happening in the `.env` file on the server

Comment (aninternetof) [reply compose/django/daphne.sh]: Oh, right. Deleting.

**Comment by aninternetof** [comment]
Huh, interesting. It doesn't seem like `gunicorn` is an installed app, so I wonder why that was there.

I can't find anything about in [the docs](https://channels.readthedocs.io/en/latest/deploying.html) that for using Daphne, so I'm just going to delete the line
